native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ArgumentHandler.java

```diff
@@ -119,11 +135,29 @@ private Object getArgumentValue(BObject argumentNode, Type parameterType) {
             return this.getUnionTypeArgument(argumentNode, (UnionType) parameterType);
         } else if (parameterType.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG) {
             return this.getArgumentValue(argumentNode, TypeUtils.getReferredType(parameterType));
+        } else if (idsList.contains(argumentNode.getStringValue(StringUtils.fromString("name")).getValue())) {
```

Comment:
Shall we define a BString const for "name" here?

Reply:
I think there is one already. `NAME_FIELD`.